.github/workflows/build.yaml

```diff
@@ -29,4 +29,18 @@ jobs:
 
       - name: Build with Maven
         run: mvn package
+      
+      # Extract version from pom.xml
+      - name: Get Version from POM
+        id: get-version
+        run: |
+          version=$(mvn help:evaluate -Dexpression=project.version -q -DforceStdout)
+          echo "version=$version" >> $GITHUB_ENV
+
+      #Publish jar if `SNAPSHOT` is present in the version
+      - name: Publish SNAPSHOT JAR
+        if: contains(env.version, '-SNAPSHOT')
```

Comment:
The other condition is that we only want to do this when code is merged to the `main` branch.

Reply:
It also looks like you need to add `packages: write` permission to the job.